document/value.go

```diff
@@ -494,27 +496,41 @@ func (v Value) ConvertToDuration() (time.Duration, error) {
 
 // IsZeroValue indicates if the value data is the zero value for the value type.
 // This function doesn't perform any allocation.
-func (v Value) IsZeroValue() bool {
+func (v Value) IsZeroValue() (bool, error) {
+	var isZeroValue bool
+
 	switch v.Type {
 	case BlobValue, TextValue:
-		return bytesutil.CompareBytes(v.V.([]byte), blobZeroValue.V.([]byte)) == 0
+		isZeroValue = bytesutil.CompareBytes(v.V.([]byte), blobZeroValue.V.([]byte)) == 0
 	case BoolValue:
-		return v.V == boolZeroValue.V
+		isZeroValue = v.V == boolZeroValue.V
 	case Int8Value:
-		return v.V == int8ZeroValue.V
+		isZeroValue = v.V == int8ZeroValue.V
 	case Int16Value:
-		return v.V == int16ZeroValue.V
+		isZeroValue = v.V == int16ZeroValue.V
 	case Int32Value:
-		return v.V == int32ZeroValue.V
+		isZeroValue = v.V == int32ZeroValue.V
 	case Int64Value:
-		return v.V == int64ZeroValue.V
+		isZeroValue = v.V == int64ZeroValue.V
 	case Float64Value:
-		return v.V == float64ZeroValue.V
+		isZeroValue = v.V == float64ZeroValue.V
 	case DurationValue:
-		return v.V == durationZeroValue.V
+		isZeroValue = v.V == durationZeroValue.V
+	case ArrayValue:
+		len, err := ArrayLength(v.V.(Array))
```

Comment:
`Length` and `ArrayLength` are not optimal for this use case. Instead of counting all the values then comparing with 0, we only need to assert that there is at least one value. O(n) -> O(1)

Reply:
Very good call! I improved this in b34ff32. Please notice the `IsEmpty` method in the `Document` interface.